pysatSpaceWeather/instruments/sw_flare.py

```diff
@@ -0,0 +1,375 @@
+                                meta.labels.name: '3 Optical Flares',
+                                meta.labels.notes: '',
+                                meta.labels.desc: '3-class Optical Flares',
+                                meta.labels.fill_val: -1,
+                                meta.labels.min_val: 0,
+                                meta.labels.max_val: 9}
+    elif tag == 'prediction':
+        for dkey in data.columns:
+            if dkey.find('Region') < 0:
+                meta[dkey] = {meta.labels.units: '',
+                              meta.labels.name: dkey,
+                              meta.labels.notes: '',
+                              meta.labels.desc: ''.join([dkey.replace('_', ' '),
+                                                         ' Probabilities']),
+                              meta.labels.fill_val: -1,
+                              meta.labels.min_val: 0,
+                              meta.labels.max_val: 100}
+            else:
+                meta[dkey] = {meta.labels.units: '',
+                              meta.labels.name: dkey,
+                              meta.labels.notes: '',
+                              meta.labels.desc: 'SWPC Solar Region Number',
+                              meta.labels.fill_val: -1,
+                              meta.labels.min_val: 1,
+                              meta.labels.max_val: 9999}
+
+    return data, meta
+
+
+def list_files(tag='', inst_id='', data_path='', format_str=None):
+    """List local solar flare data files.
+
+    Parameters
+    ----------
+    tag : str
+        Instrument tag, accepts any value from `tags`. (default='')
+    inst_id : str
+        Instrument ID, not used. (default='')
+    data_path : str
+        Path to data directory. (default='')
+    format_str : str or NoneType
+        User specified file format.  If None is specified, the default
+        formats associated with the supplied tags are used. (default=None)
+
+    Returns
+    -------
+    out_files : pysat._files.Files
+        A class containing the verified available files
+
+    Note
+    ----
+    Called by pysat. Not intended for direct use by user.
+
+    """
+
+    if tag == 'prelim':
+        # Files are by year (and quarter)
+        if format_str is None:
+            format_str = ''.join(['flare_prelim_{year:04d}_{month:02d}',
+                                  '_v{version:01d}.txt'])
+        out_files = pysat.Files.from_os(data_path=data_path,
+                                        format_str=format_str)
+
+        if not out_files.empty:
+            # Set each file's valid length at a 1-day resolution
+            orig_files = out_files.sort_index().copy()
+            new_files = list()
+
+            for orig in orig_files.items():
+                # Version determines each file's valid length
+                version = int(orig[1].split("_v")[1][0])
+                doff = pds.DateOffset(years=1) if version == 2 \
```

Comment:
I believe we moved to dt.timedelta.

Reply:
We did unless time increments not handled by timedelta are used.  This is one such instance.